cpp/src/plasma/CMakeLists.txt

```diff
@@ -123,6 +123,18 @@ endif()
 
 add_executable(plasma_store store.cc)
 target_link_libraries(plasma_store plasma_static ${PLASMA_LINK_LIBS})
+SET_TARGET_PROPERTIES(plasma_store PROPERTIES
+  LINK_FLAGS "-Wl,--as-needed")
```

Comment:
Can you use https://cmake.org/cmake/help/v3.0/module/CheckCXXCompilerFlag.html and add this flag globally? If the compiler supports it, we should use it everywhere.